Add CSP header page

sbreker opened the pull request:
Add security section page detailing CSP header functionality.

sbreker committed:
Add CSP header page

Add security section page detailing CSP header functionality.

fiver-watson commented on admin-manual/security/csp-headers.rst:
```diff
@@ -0,0 +1,263 @@
+===============================
```
Would be good to add an anchor to the main title page - we tend to link throughout the docs to anchors rather than pages, so if we move pages around something doesn't break. I would suggest: 

```
.. _security-csp-headers:
```

You will also need this ref to be able to link on the docs homepage, which it looks like you forgot to add in this PR!  (i.e. `atom-docs/index.rst`, under the security section of the Admin part)

sbreker replied:
Added.

fiver-watson commented on admin-manual/security/csp-headers.rst:
nitpick: should it be "lacks" rather than lack here? hard to keep track of...

sbreker replied:
Yes - that sounds better - fixed.

fiver-watson commented on admin-manual/security/csp-headers.rst:
might be good to spell this out the first time, i.e. "protects against cross-site scripting (XSS) attacks and..."

sbreker replied:
Got this covered on line 13:

https://github.com/artefactual/atom-docs/pull/243/files#diff-0645c07fc50392ff698bc7dafe5f38576b3fd01a4c5f17ab6ee6c6eca8efe8b7R13

sbreker replied:
Added.

fiver-watson commented on admin-manual/security/csp-headers.rst:
I've not see this used before, did you generate the preview files when writing these and ensure that it renders correctly? If yes, great! 

I've always done it by first specifying it's a code block, followed by the specific language / format used - if Sphinx knows it, then it will properly color the syntax, etc - like this: 

```
.. code-block:: mermaid

```

That said, if this short-hand works in the local renders, then nevermind!

sbreker replied:
I have tested this locally and it seems to work. I've added the appropriate references for this to work in a second commit.

sbreker replied:
![image](https://github.com/artefactual/atom-docs/assets/3812460/d092807a-c11c-4364-a6d9-b2b58b6428f1)

fiver-watson replied:
yep, looks good! thanks for double-checking too :D

fiver-watson commented on admin-manual/security/csp-headers.rst:
I would suggest linking to the instructions for restarting PHP-FPM, since it changes with each PHP version. i.e. 

```
Updating these settings will require restarting :ref:`PHP-FPM <troubleshooting-restart-php-fpm>`.
```

sbreker replied:
Good idea! Added...

fiver-watson commented on admin-manual/security/csp-headers.rst:
I'd suggest putting this in an IMPORTANT admonition at the bottom of this section, since a) we want to emphasize it now, but b) in 2.9 and later we can remove that mention, since there will only be B5 themes. i.e.

```
.. IMPORTANT::

   CSP headers will only be applied toa response if a Bootstrap 5 based theme is in use. See: 

   * :ref:`customization-theming`
   * :ref:`themes`
```

sbreker replied:
Good idea! Added...

fiver-watson commented on admin-manual/security/csp-headers.rst:
again, good idea to add a sub-heading anchor / ref here, so we can link directly to this. Dealer's choice.

sbreker replied:
Added `.. _csp-custom-themes:`

fiver-watson commented on admin-manual/security/csp-headers.rst:
Do we need to reference a specific version here (2.8)? Just thinking that we will need to remember to update this in the future...

sbreker replied:
Removed. This is in the 2.8 docs, so it will be understood to look in AtoM 2.8's app.yml.

fiver-watson commented on admin-manual/security/csp-headers.rst:
Also useful to have anchors in at least the first-level subheadings, so we can link directly to them when relevant in responding to client emails, forum posts, etc. 

I'd suggest something like `.. _csp-headers-scenario:` here maybe, or `-overview`, etc?

sbreker replied:
Added `_csp-overview`

fiver-watson commented on admin-manual/security/csp-headers.rst:
i'd suggest prefacing this anchor with `csp-` so we can tell what the anchor relates to at a glance, and all CSP anchors are similarly named? (and generally all subheadings on this page, minus the page header one, which gets the `security-` prefix to match other pages in the section).... e.g. 

```
.. _csp-allow-inline-sources:
```

sbreker replied:
Added

fiver-watson commented on admin-manual/security/csp-headers.rst:
Same feedback, add an anchor plz :D

sbreker replied:
Added

fiver-watson commented on admin-manual/security/csp-headers.rst:
Nice!

sbreker replied:
I think this was from your security doc! :+1:

fiver-watson commented on requirements.txt:
```diff
@@ -1,2 +1,3 @@
 sphinx
 sphinx_rtd_theme
+sphinxcontrib-mermaid
```
oh cool, i guess that this add support for the mermaid block you added above! great, if it's tested then you can ignore my comments there

sbreker replied:
:+1:

fiver-watson reviewed (changes requested):
Hi @sbreker , this looks great! 

You forgot to add a link to it to the general `index.rst` landing page for all the docs, and I've requested some subheader anchors be added, but overall this content looks awesome. Thanks!

fiver-watson commented on admin-manual/security/csp-headers.rst:
nitpick: just saw this tiiiiiiny typo here (toa instead of "to a")

sbreker replied:
Nice catch! Fixed :+1:

fiver-watson reviewed (approved):
I caught one minor typo I missed on the first round, but it's a minor issue. Otherwise, everything else looks great! Thanks Steve!

fiver-watson reviewed (approved):
:+1:

fiver-watson commented:
@sbreker did this never get merged? or is this an older version and the work got merged elsewhere? just wondering if some cleanup is needed here...

anvit committed:
Fix typo in reference label

sbreker committed:
Replace CSP Mermaid diagram with image

sbreker committed:
Update the CSP headers page

Added note about updating the default header type. Updated directive to
include the Google integration changes. Additional detail added to CSP
implementation section.

anvit reviewed (approved):
Looks good! Thanks @sbreker !

fiver-watson reviewed (approved):
Looks great!
